File: 2022/python/day01.py
def get_data():
    with open("../input/day1.txt", "r") as handle:
        lines = handle.readlines()

    sum_calories = 0
    data = []
    for line in lines:
        if line == "\n":
            data.append(sum_calories)
            sum_calories = 0
        else:
            sum_calories += int(line.strip("\n"))
    return data


def part1():
    data = get_data()
    print(f"day1-part1: {max(data)}")


def part2():
    data = get_data()
    sum_calories = sum(sorted(data)[-3:])
    print(f"day1-part2: {sum_calories}")


if __name__ == "__main__":
    part1()
    part2()

File: 2022/rust/src/day01.rs
use std::fs;

fn main() {
    let input = fs::read_to_string("../input/day1.txt").unwrap();
    let mut data: Vec<usize> = vec![];
    let mut current: usize = 0;

    for line in input.lines() {
        if line.len() == 0 {
            data.push(current);
            current = 0;
        } else {
            current += line.trim().parse::<usize>().unwrap();
        }
    }

    data.sort();

    println!("day1-part1: {}", data[data.len() - 1]);
    println!(
        "day1-part2: {}",
        data[data.len() - 1] + data[data.len() - 2] + data[data.len() - 3]
    );
}
